File: Solutions/Python/rotate_array.py
"""
    [MEDIUM]
    189. Rotate Array

    Concepts:
    - array
    - math

    Stats:
        Runtime | 5 ms      [Beats 53.43%]
        Memory  | 25.62 MB  [Beats 32.20%]
"""

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k %= n
        
        cpy = nums.copy()

        for i, num in enumerate(cpy):
            nums[i] = cpy[(i - k) % n]
        

File: Solutions/Cpp/rotate_array.cpp
/*
    [MEDIUM]
    189. Rotate Array

    Concepts:
    - array

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 29.61 MB  [Beats 39.40%]
*/

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + k);
        reverse(nums.begin() + k, nums.end());
    }
};